internal/controller/stages/stages.go

```diff
@@ -1005,8 +1005,32 @@ func (r *reconciler) syncPromotions(
 	// current state of the Stage.
 	slices.SortFunc(promotions, kargoapi.ComparePromotionByPhaseAndCreationTime)
 
+	// The latest Promotion is the first element in the sorted list.
+	latestPromo := promotions[0]
```

Comment:
Is latestPromo misleading? If I understand the sort order correctly, it's running promos first, then non-terminal, then terminal. And each of those categories are internally sorted by ULID, which implicitly sorts by creation date. Have I got that right? IF I have got that right, then `promotions[0]` doesn't feel "latest" (most recently created?) to me.

Reply:
I believe the logic is sound. It's just the name I found confusing.

Reply:
What about something like `priorityPromo`? As, I believe as sorted, `promotions[0]` _is_ the highest priority?

Reply:
`highestPrioPromo` then maybe?